src/common/transformations/tests/op_conversions/scaled_dot_product_decomposition_test.cpp

```diff
@@ -0,0 +1,159 @@
+    const auto key = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale_shape);
+    const auto value = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale_shape);
+    const auto attention_mask = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale_shape);
+    const auto scale = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale_shape);
+    const auto casual = false;
+    {
+        const auto scaled_dot_product_attention =
+            std::make_shared<ov::op::v13::ScaledDotProductAttention>(query, key, value, attention_mask, scale, casual);
+
+        model = std::make_shared<ov::Model>(NodeVector{scaled_dot_product_attention},
+                                            ParameterVector{query, key, value, attention_mask, scale});
+        manager.register_pass<ov::pass::ScaledDotProductAttentionDecomposition>();
+    }
+
+    {
+        const auto scaled_dot_product_attention =
+            scaled_dot_product_attention_decomposition(query, key, value, attention_mask, scale, casual);
+        model_ref = std::make_shared<ov::Model>(NodeVector{scaled_dot_product_attention},
+                                                ParameterVector{query, key, value, attention_mask, scale});
+    }
+}
+
+TEST_F(TransformationTestsF, ScaledDotProductAttentionDecompositionDynamic) {
+    const PartialShape query_key_value_mask_scale_shape{-1, -1, -1};
+
+    const auto query = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale_shape);
+    const auto key = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale_shape);
+    const auto value = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale_shape);
+    const auto attention_mask = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale_shape);
+    const auto scale = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale_shape);
+    const auto casual = false;
+    {
+        const auto scaled_dot_product_attention =
+            std::make_shared<ov::op::v13::ScaledDotProductAttention>(query, key, value, attention_mask, scale, casual);
+
+        model = std::make_shared<ov::Model>(NodeVector{scaled_dot_product_attention},
+                                            ParameterVector{query, key, value, attention_mask, scale});
+        manager.register_pass<ov::pass::ScaledDotProductAttentionDecomposition>();
+    }
+
+    {
+        const auto scaled_dot_product_attention =
+            scaled_dot_product_attention_decomposition(query, key, value, attention_mask, scale, casual);
+        model_ref = std::make_shared<ov::Model>(NodeVector{scaled_dot_product_attention},
+                                                ParameterVector{query, key, value, attention_mask, scale});
+    }
+}
+
```

Comment:
Maybe add few more tests that would ensure that supported broadcasts of batches and other corresponding elements work as expected?

Reply:
Added 2 more tests with crazy shapes to verify Broadcast feasibility